README.md

```diff
@@ -28,16 +28,47 @@ yarn add ethers-proxies
 
 ## How to use
 
-The function needs an ethers provider it uses to run a set of checks against the given address.
+The function requires an [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193) compatible `request` function that it uses to make JSON-RPC requests to run a set of checks against the given address.
 It returns a promise that resolves to the proxy target address, i.e., the address of the contract implementing the logic.
 The promise resolves to `null` if no proxy can be detected.
 
+### Ethers with an EIP1193Bridge
 ```ts
-import { InfuraProvider } from '@ethersproject/providers'
+import { JsonRpcProvider } from '@ethersproject/providers'
+import { Eip1193Bridge } from "@ethersproject/experimental";
 import detectProxyTarget from 'ethers-proxies'
 
-const provider = new InfuraProvider(1, process.env.INFURA_API_KEY)
-const target = await detectProxyTarget('0xA7AeFeaD2F25972D80516628417ac46b3F2604Af', provider)
+const provider = new JsonRpcProvider(`http://localhost:8545`)
+const signer = provider.getSigner(0)
+const eip1193Provider = new Eip1193Bridge(signer, provider)
+const requestFunc = eip1193Provider.request.bind(eip1193Provider)
+
+const target = await detectProxyTarget('0xA7AeFeaD2F25972D80516628417ac46b3F2604Af', requestFunc)
+console.log(target)  // logs "0x4bd844F72A8edD323056130A86FC624D0dbcF5b0"
```

Comment:
Let's just remove this example. It's unnecessarily complex, since you could just use `provider.send.bind(provider)` directly instead of going through the bridge.